src/system_database.ts

```diff
@@ -895,11 +1021,30 @@ export class PostgresSystemDatabase implements SystemDatabase {
     await this.recordWorkflowStatusChange(workflowID, status, { resetRecoveryAttempts });
   }
 
+  setWFCancelMap(workflowID: string) {
+    if (this.runningWorkflowMap.has(workflowID)) {
+      this.workflowCancellationMap.set(workflowID, true);
+    }
+    this.cancelWakeupMap.callCallbacks(workflowID);
+  }
+
+  clearWFCancelMap(workflowID: string) {
+    if (this.workflowCancellationMap.has(workflowID)) {
+      this.workflowCancellationMap.delete(workflowID);
+    }
+  }
+
   async cancelWorkflow(workflowID: string): Promise<void> {
     const client = await this.pool.connect();
     try {
       await client.query('BEGIN');
 
+      await client.query(
+        `INSERT INTO ${DBOSExecutor.systemDBSchemaName}.workflow_cancel(workflow_id)
+         VALUES ($1)`,
```

Comment:
Probably need a `ON CONFLICT DO NOTHING` here. For example, a client may call `cancelWorkflow` multiple times but they shouldn't fail.